storage/src/main/java/tech/pegasys/artemis/storage/ChainStorageClient.java

```diff
@@ -46,28 +50,33 @@
 /** This class is the ChainStorage client-side logic */
 public class ChainStorageClient implements ChainStorage {
   private static final ALogger STDOUT = new ALogger("stdout");
-  static final ALogger LOG = new ALogger(ChainStorageClient.class.getName());
-  private Store store;
-  protected EventBus eventBus;
+
+  protected final EventBus eventBus;
   protected final ConcurrentHashMap<Bytes32, Bitlist> processedAttestationsBitlistMap =
       new ConcurrentHashMap<>();
   protected final ConcurrentHashMap<Bytes32, UnsignedLong> processedAttestationsToEpoch =
       new ConcurrentHashMap<>();
   private final ConcurrentHashMap<Bytes32, Attestation> unprocessedAttestationsMap =
       new ConcurrentHashMap<>();
+  private final NavigableMap<UnsignedLong, Bytes32> slotToCanonicalBlockRoot =
```

Comment:
This may be a question for another PR, but we need to limit how far back we maintain this mapping (at least for this in-memory version).  `Store.cleanStoreUntilSlot` removes old blocks from the store which would be the point to also remove this mapping.  But at the moment we don't appear to call `cleanStoreUntilSlot` from production code. https://pegasys1.atlassian.net/browse/BC-33 covers doing this pruning so we could just add a note there to ensure the mapping is also pruned.